WcaOnRails/app/controllers/api/v0/roles_controller.rb

```diff
@@ -145,36 +202,84 @@ def show
   end
 
   def update
-    user_id = params.require(:userId)
-    delegate_status = params.require(:delegateStatus)
-    region_id = params.require(:regionId)
-    location = params.require(:location)
+    id = params.require(:id)
 
-    user = User.find(user_id)
-    if delegate_status == "senior_delegate"
-      senior_delegate_id = nil
-      User.where(region_id: region_id).where.not(id: user_id).update_all(senior_delegate_id: user_id)
+    if id == "dummyRoleId" # dummyRoleId is temporarily reserved for delegate changes.
```

Comment:
Even if it's only temporary, it might be worth extracting it into a constant.
If the frontend needs to know about this constant, you might even consider exposing it via `lib/wca-data.js.erb` to have a single source of truth and avoid typos in the string literal costing you 30 minutes of debugging :)